sonar-db/src/main/java/org/sonar/db/source/FileSourceDao.java

```diff
@@ -108,6 +119,29 @@ public FileSourceDto selectTest(String fileUuid) {
     }
   }
 
+  public Optional<CloseableIterator<DbFileSources.Line>> selectSourceLines(DbSession dbSession, String fileUuid) {
+    Connection connection = dbSession.getConnection();
+    PreparedStatement pstmt = null;
+    ResultSet rs = null;
+    try {
+      pstmt = connection.prepareStatement("SELECT binary_data FROM file_sources WHERE file_uuid=? AND data_type=?");
+      pstmt.setString(1, fileUuid);
+      pstmt.setString(2, Type.SOURCE);
+      rs = pstmt.executeQuery();
+      if (rs.next()) {
+        InputStream stream = new LZ4BlockInputStream(rs.getBinaryStream(1));
+        CloseableIterator<DbFileSources.Line> linesIterator = Protobuf.readStream(stream, DbFileSources.Line.PARSER);
+        return Optional.of(CloseableIterator.wrap(linesIterator, rs, pstmt, connection));
+      } else {
+        DbUtils.closeQuietly(connection, pstmt, rs);
+        return Optional.absent();
+      }
+    } catch (Throwable e) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1181)